ui/pages/confirmations/components/confirm/nav/nav.tsx

```diff
@@ -26,68 +25,46 @@ import {
   TextColor,
   TextVariant,
 } from '../../../../../helpers/constants/design-system';
-import {
-  CONFIRM_TRANSACTION_ROUTE,
-  SIGNATURE_REQUEST_PATH,
-} from '../../../../../helpers/constants/routes';
 import { useI18nContext } from '../../../../../hooks/useI18nContext';
-import { pendingConfirmationsSortedSelector } from '../../../../../selectors';
 import { rejectPendingApproval } from '../../../../../store/actions';
-import { useConfirmContext } from '../../../context/confirm';
 import { useQueuedConfirmationsEvent } from '../../../hooks/useQueuedConfirmationEvents';
-import { isCorrectSignatureApprovalType } from '../../../../../../shared/lib/confirmation.utils';
+import { useConfirmationNavigation } from '../../../hooks/useConfirmationNavigation';
+import { useConfirmContext } from '../../../context/confirm';
+
+export type NavProps = {
+  confirmationId?: string;
+};
 
-const Nav = () => {
-  const history = useHistory();
+export const Nav = ({ confirmationId }: NavProps) => {
   const t = useI18nContext();
   const dispatch = useDispatch();
 
-  const { currentConfirmation } = useConfirmContext();
-
-  const pendingConfirmations = useSelector(pendingConfirmationsSortedSelector);
+  const { confirmations, count, getIndex, navigateToIndex } =
+    useConfirmationNavigation();
 
-  const currentConfirmationPosition = useMemo(() => {
-    if (pendingConfirmations?.length <= 0 || !currentConfirmation) {
-      return 0;
-    }
-    return pendingConfirmations.findIndex(
-      ({ id }) => id === currentConfirmation.id,
-    );
-  }, [currentConfirmation, pendingConfirmations]);
+  const position = getIndex(confirmationId);
 
-  const onNavigateToTransaction = useCallback(
-    (pos: number) => {
-      const nextConfirmation =
-        pendingConfirmations[currentConfirmationPosition + pos];
-      // todo: once all signature request pages are ported to new designs
-      // SIGNATURE_REQUEST_PATH from path below can be removed
-      // In new routing all confirmations will support
-      // "/confirm-transaction/<confirmation_id>"
-      history.replace(
-        `${CONFIRM_TRANSACTION_ROUTE}/${nextConfirmation.id}${
-          isCorrectSignatureApprovalType(nextConfirmation.type as ApprovalType)
-            ? SIGNATURE_REQUEST_PATH
-            : ''
-        }`,
-      );
+  const onNavigateButtonClick = useCallback(
+    (change: number) => {
+      navigateToIndex(position + change);
     },
-    [currentConfirmationPosition, pendingConfirmations],
+    [position, navigateToIndex],
   );
 
   const onRejectAll = useCallback(() => {
-    pendingConfirmations.forEach((conf) => {
+    confirmations.forEach((conf) => {
       dispatch(
         rejectPendingApproval(
```

Comment:
Hmm, not confident that we can treat all confirmations as equal when rejecting.

For instance, some templates for Snaps confirmations have special logic for handling cancel: https://github.com/MetaMask/metamask-extension/blob/b078721197f5d392333fcd60bf4f4cbf875fa410/ui/pages/confirmations/confirmation/templates/snaps/snap-confirmation/snap-confirmation.js#L13-L14
https://github.com/MetaMask/metamask-extension/blob/b078721197f5d392333fcd60bf4f4cbf875fa410/ui/pages/confirmations/confirmation/templates/snaps/snap-alert/snap-alert.js#L7-L10

Reply:
That's a pain, but makes sense. Ideally, we'd maintain the ability to reject from the wallet, and the snaps would handle accordingly as they wish.

 I'll clarify with the team what we want to do UX wise.

Reply:
> Ideally, we'd maintain the ability to reject from the wallet, and the snaps would handle accordingly as they wish.

No matter what we'll need some form of backwards compatibility. As it stands, this would be a breaking change for how Snaps dialogs function (via change in expected return value).

Reply:
It would feel very weird to be unable to cancel snaps requests in a queue. Is there something we can do so users are able to cancel those, @FrederikBolding ? 

Would this also mean that if a snap never closes a confirmation the user could have this confirmation hanging in the queue forever because they are not able to close it themselves?

Reply:
Oh no, I think there has been a misunderstanding. My comment is only about making sure that when we reject Snaps confirmations, we give the Snap the value back that it expects. We definitely want to be able to cancel the requests, we just need to think about how to do it on the technical side!

Reply:
Done, and from popup close also.